lib/ansible/modules/cloud/azure/azure_rm_sqldatabase.py

```diff
@@ -0,0 +1,483 @@
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.server_name = None
+        self.name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False, state=dict())
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMDatabases, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                               supports_check_mode=True,
+                                               supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif key == "tags":
+                self.parameters["tags"] = kwargs[key]
+            elif key == "location":
+                self.parameters["location"] = kwargs[key]
+            elif key == "collation":
+                self.parameters["collation"] = kwargs[key]
+            elif key == "create_mode":
+                self.parameters["create_mode"] = kwargs[key]
+            elif key == "source_database_id":
+                self.parameters["source_database_id"] = kwargs[key]
+            elif key == "source_database_deletion_date":
+                self.parameters["source_database_deletion_date"] = kwargs[key]
+            elif key == "restore_point_in_time":
+                self.parameters["restore_point_in_time"] = kwargs[key]
+            elif key == "recovery_services_recovery_point_resource_id":
+                self.parameters["recovery_services_recovery_point_resource_id"] = kwargs[key]
+            elif key == "edition":
+                self.parameters["edition"] = kwargs[key]
+            elif key == "max_size_bytes":
+                self.parameters["max_size_bytes"] = kwargs[key]
+            elif key == "requested_service_objective_id":
+                self.parameters["requested_service_objective_id"] = kwargs[key]
+            elif key == "requested_service_objective_name":
+                self.parameters["requested_service_objective_name"] = kwargs[key]
+            elif key == "elastic_pool_name":
+                self.parameters["elastic_pool_name"] = kwargs[key]
+            elif key == "read_scale":
+                self.parameters["read_scale"] = kwargs[key]
+            elif key == "sample_name":
+                self.parameters["sample_name"] = kwargs[key]
+            elif key == "zone_redundant":
+                self.parameters["zone_redundant"] = kwargs[key]
+
+        old_response = None
+        results = dict()
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
```

Comment:
As we talked about in the WG yesterday, please set the API version in the module and pass it along here.